opal/mca/common/ucx/common_ucx_wpool.h

```diff
@@ -105,9 +105,14 @@ typedef enum {
     OPAL_COMMON_UCX_MEM_MAP
 } opal_common_ucx_mem_type_t;
 
+typedef struct {
+    void *external_req;
```

Comment:
Add:
* user callback
* winfo ptr is to be added in subsequent commit introducing fine-grained locking